airflow/providers/apache/kafka/operators/await_message.py

```diff
@@ -0,0 +1,107 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Any, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.apache.kafka.triggers.await_message import AwaitMessageTrigger
+
+VALID_COMMIT_CADENCE = {"never", "end_of_batch", "end_of_operator"}
+
+
+class AwaitKafkaMessageOperator(BaseOperator):
+    """AwaitKafkaMessageOperator An Airflow operator that defers until a specific message is published to Kafka.
+
+    The operator creates a consumer that reads the Kafka log until it encounters a positive event.
+
+    The behavior of the consumer for this trigger is as follows:
+    - poll the Kafka topics for a message
+    - if no message returned, sleep
+    - process the message with provided callable and commit the message offset
+    - if callable returns any data, raise a TriggerEvent with the return data
+    - else continue to next message
+    - return event (as default xcom or specific xcom key)
+
+    :param topics: Topics (or topic regex) to use for reading from
+    :param apply_function: The functoin to apply to messages to determine if an event occurred. As a dot
+    notation string.
```

Comment:
```suggestion
    :param apply_function: The functoin to apply to messages to determine if an event occurred. As a dot
        notation string.
```
I forget how picky Sphinx is, but IIRC if the param description goes beyond the first line, each subsequent line needs to be indented. You can [build the docs locally with Breeze](https://github.com/apache/airflow/blob/main/BREEZE.rst#building-the-documentation) and confirm.